test/e2e/kcp_adoption.go

```diff
@@ -96,7 +97,7 @@ func KCPAdoptionSpec(ctx context.Context, inputGetter func() KCPAdoptionSpecInpu
 			// pass the clusterctl config file that points to the local provider repository created for this test,
 			ClusterctlConfigPath: input.ClusterctlConfigPath,
 			// select template
-			Flavor: "kcp-adoption",
+			Flavor: pointer.StringDeref(input.Flavor, "kcp-adoption"),
```

Comment:
I can see how it might be good to be able to customize the flavors for most of these tests, but for some of them like kcp-adoption it expects a very specific template (this one requires kcp-adoption.step1 and .step2 for example) and so it's not like flavors and test specs are really interchangeable. How can we make sure the requirements of the test are met in the template without necessarily enforcing the name?

Reply:
@CecileRobertMichon To give an idea of why this is desirable for me. I'm working on adding support to CAPP to allow someone to optionally use the cloud-provider integration or not and I want to more easily be able to test all of these scenarios for both cloud-provider integration enabled and disabled without having to run a separate test suite using a different e2e configuration.

If I can override the flavor, then I can generate `cluster-template-kcp-adoption.yaml` and `cluster-template-kcp-adoption-no-cloud-provider.yaml` templates instead of having to invoke the tests twice with different e2e configs that take different inputs and create the same `cluster-template-kcp-adoption.yaml` template.

KCP adoption is probably the worst example to use, though, since we aren't planning on actually using it for testing or supporting it. It just seemed appropriate to make things consistent.

Other tests that require specific configuration (or more specifically have hard coded flavors), such as the kcp-scale-in,  remediation, node drain, and upgrade tests are applicable for our needs.

Reply:
That's a great idea, I'll go ahead and add some comments.

Reply:
Added documentation for all places where a pointer is used for Flavor, documenting the requirements for the tests to the best of my understanding of them. I also added what values are used if not specified.